src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -797,23 +797,28 @@ private string GenerateCtorParamMetadataInitFunc(TypeGenerationSpec typeGenerati
 private static {JsonParameterInfoValuesTypeRef}[] {typeGenerationSpec.TypeInfoPropertyName}{CtorParamInitMethodNameSuffix}()
 {{
     {JsonParameterInfoValuesTypeRef}[] {parametersVarName} = new {JsonParameterInfoValuesTypeRef}[{paramCount}];
-    {JsonParameterInfoValuesTypeRef} info;
 ");
 
                 for (int i = 0; i < paramCount; i++)
                 {
                     ParameterInfo reflectionInfo = parameters[i].ParameterInfo;
 
+                    string parameterTypeRef = reflectionInfo.ParameterType.GetCompilableName();
+
+                    object? defaultValue = reflectionInfo.GetDefaultValue();
+                    string defaultValueAsStr = defaultValue == null
+                        ? $"default({parameterTypeRef})"
```

Comment:
The `GetParamDefaultValueAsString` already seems to contain handling logic for null "default values". Why do we need to special case that here?

Reply:
I've moved the special casing into `GetParamDefaultValueAsString`. Special casing is needed to provide a `default` value for value-type params and avoid `InvalidCastException` from `null` to those types in the generated ctor invocation logic.